lib/experimental/new_embed.dart

```diff
@@ -645,6 +647,111 @@ class FlashBox {
   void hide() {
     _element.setAttr('hidden');
   }
+
+  void show() {
+    _element.clearAttr('hidden');
+  }
+}
+
+class AnalysisResultsController {
+  static const String _noIssuesMsg = 'no issues';
+  static const String _hideMsg = 'hide';
+  static const String _showMsg = 'show';
+
+  static const Map<String, List<String>> _classesForType = {
+    'info': ['issuelabel', 'info'],
+    'warning': ['issuelabel', 'warning'],
+    'error': ['issuelabel', 'error'],
+  };
+
+  DElement flash;
+  DElement message;
+  DElement toggle;
+  bool _hidden = true;
+
+  final StreamController<AnalysisIssue> _onClickController =
+      StreamController.broadcast();
+  Stream<AnalysisIssue> get onIssueClick => _onClickController.stream;
```

Comment:
Is this StreamController necessary? Could the constructor instead accept a callback to be invoked when an issue is clicked (which is then used directly on line 730)?

Reply:
This would restrict the API to only supporting one listener. Is that what we want?

Reply:
Fair point.